python/ray/data/_internal/batcher.py

```diff
@@ -49,11 +49,12 @@ class Batcher(BatcherInterface):
     # zero-copy views, we sacrifice what should be a small performance hit for better
```

Comment:
This comment seems no longer true now.

Reply:
This is a good point, this actually might make the batcher a good bit more expensive for the `.map_batches()` case, and there being multiple batches per block is also the common case, so this perf/memory stability hit could be significant. I'll try to think of another approach that avoids this!

Reply:
Ultimately reverting back to the old zero-copy slice behavior, instead we always copy the batch before returning it. Although this will be unnecessary in a few cases (e.g. pandas batches that already required block concatenation to produce them), it should avoid the pathological nested copying that the old implementation would have run into.